examples/decoding/plot_receptive_field.py

```diff
@@ -151,4 +158,94 @@
+                    estimator=1e4, scoring='corrcoef', patterns=True)
+# We'll have (tmax - tmin) * sfreq delays
+# and an extra 2 delays since we are inclusive on the beginning / end index
+n_delays = int((tmax - tmin) * sfreq) + 2
+
+n_splits = 3
+cv = KFold(n_splits)
+
+# Iterate through splits, fit the model, and predict/test on held-out data
+coefs = np.zeros((n_splits, n_channels, n_delays))
+patterns = coefs.copy()
+scores = np.zeros((n_splits,))
+for ii, (train, test) in enumerate(cv.split(speech)):
+    print('split %s / %s' % (ii + 1, n_splits))
+    sr.fit(Y[train], speech[train])
+    scores[ii] = sr.score(Y[test], speech[test])[0]
+    # coef_ is shape (n_outputs, n_features, n_delays). we have 128 features
+    coefs[ii] = sr.coef_[0, :, :]
+    patterns[ii] = sr.patterns_[0, :, :]
+times = sr.delays_ / float(sr.sfreq)
+
+# Average scores and coefficients across CV splits
+mean_coefs = coefs.mean(axis=0)
+mean_patterns = patterns.mean(axis=0)
+mean_scores = scores.mean(axis=0)
+max_coef = np.abs(mean_coefs).max()
+max_patterns = np.abs(mean_patterns).max()
+
+# Visualize stimulus reconstruction
+# ---------------------------------
+#
+# To get a sense of our model performance, we can plot the actual and predicted
+# stimulus envelopes side by side.
+
+y_pred = sr.predict(Y[test])
+time = np.linspace(0, 2., 5 * int(sfreq))
+fig, ax = plt.subplots(figsize=(8, 4))
+ax.plot(time, speech[test][sr.valid_samples_][:int(5 * sfreq)],
+        color='grey', lw=2, ls='--')
+ax.plot(time, y_pred[sr.valid_samples_][:int(5 * sfreq)], color='r', lw=2)
+ax.legend([lns[0], ln1[0]], ['Envelope', 'Reconstruction'], frameon=False)
+ax.set(title="Stimulus reconstruction")
+ax.set_xlabel('Time (s)')
+mne.viz.tight_layout()
+
+# Investigate model coefficients
+# ------------------------------
+#
+# Finally, we will look at how the linear coefficients (sometimes
+# referred to as beta values) are distributed across the scalp. We will
+# recreate `figure 5`_ from [1]_. The backward model weights reflect the
+# channels that contribute most toward reconstructing the stimulus signal, but
+# are not directly interpretable in a neurophysiological sense. Here we also
+# look at the inversed model weights, which are easier to understand [2]_.
+
+time_plot = (-.140, -.125)  # To average between two timepoints.
+ix_plot = np.arange(np.argmin(np.abs(time_plot[0] - times)),
+                    np.argmin(np.abs(time_plot[1] - times)))
+fig, ax = plt.subplots(1, 2)
+mne.viz.plot_topomap(np.mean(mean_coefs[:, ix_plot], axis=1),
+                     pos=info, axes=ax[0], show=False,
+                     vmin=-max_coef, vmax=max_coef)
+ax[0].set(title="Backward model coefficients")
+
+mne.viz.plot_topomap(np.mean(mean_patterns[:, ix_plot], axis=1),
+                     pos=info, axes=ax[1],
+                     show=False, vmin=-max_patterns, vmax=max_patterns)
+ax[1].set(title="Forward-transformed coefficients")
```

Comment:
Can you add the "for delay ..." so there is a better correspondence with the earlier section?